exc2/src/harmonicFunctions.py

```diff
@@ -0,0 +1,95 @@
+            if direction_start is direction_end:
+                self._periodicity = t-self._t_start
+
+    def check_global_minima(self,value, t, index):
+        """[summary]
+            Method that finds the global (gurantueed in this case due to periodicity) minima of the function. Also funds the period.
+        Args:
+            value ([float]): [Value of h(t) at t]
+            t ([gloat]): [Current t]
+            index ([int]): [Index of current t and h(t)]
+        """        
+        if value < self._global_minima_value: # If the value is smaller we have a new minima
+            self._global_minima_value = value 
+        elif value == self._global_minima_value: # If value is equal we're at a new minima pos
+            if self._global_minima_t is not None and self._periodicity is None: # If periodicity is None then we need to set it
+                self._periodicity = t-self._global_minima_t
+        else:    
+            if self._index_of_first_minima is None:
+                self._index_of_first_minima = index-1 # minus 1 bcs we have started to increase again.
+                self._global_minima_t = t-self._step_size
+
+    def make_harmonic(self):
+        """[summary]
+            Method that creates the function of the object
+        Returns:
+            [{step_size:float, period:float, data:{t:[float], h_t:[float]}}]: [A dict containing the function values, time t, period and step size]
+        """        
+        function_values_and_time = {'t': [], 'h_t':[]}
+        values_of_t = np.arange(self._t_start, self._t_stop+self._step_size, self._step_size) #for-loop instead of a lamda expression for easier interpretation
+        for index, t in enumerate(values_of_t):
+            function_value_at_t = 3*math.pi*math.exp(-(self.lambda_method(t)))
+            if index > 1 and self._periodicity is None:
+                self.find_period(function_values_and_time['h_t'][0],function_values_and_time['h_t'][1], \
+                                function_value_at_t, function_values_and_time['h_t'][index-1],t)
+            #self.check_global_minima(function_value_at_t, t, index)
+            function_values_and_time['t'].append(t)
+            function_values_and_time['h_t'].append(function_value_at_t)
+        if self._periodicity is None: # This is necessary bcs we might look at an interval that is less than a period.
+            self._periodicity = index
+        return {'step_size': self._step_size, 'period': self._periodicity, 'data':function_values_and_time}
+    
+    def lambda_method(self,t):
+        """[summary]
+            Method that performs the mathematics of function lamda described in the excercise.
+        Args:
+            t ([float]): [Time t]
+
+        Returns:
+            [float]: [value of lamda at time t]
+        """        
+        return 5*math.sin(2*math.pi*1*t) # I don't see the value of 1 here but this is how lamda is defined in the exercise.
```

Comment:
```suggestion
        return 5*math.sin(2*math.pi*t) # I don't see the value of 1 here but this is how lamda is defined in the exercise.
```

It's fine, you can remove the "1"